src/neural/xla/onnx2hlo.cc

```diff
@@ -381,8 +446,79 @@ class Onnx2HloConverter {
     return {builder_.Dot(lhs, rhs, dn)};
   }
 
+  std::vector<HloFlow> OpGlobalAveragePool(const pblczero::NodeProto& node) {
+    CheckKnownAttributes(node, 1, {});
+    auto* lhs = GetInput(node, 0);
+    if (lhs->shape().dimensions_size() < 2) {
+      throw Exception("GlobalAveragePool requires at least 2D input");
+    }
+    HloFlow zero = MakeScalar(0, lhs->shape().element_type());
+    std::vector<int64_t> reduction_dims;
+    size_t num_elements = 1;
+    for (size_t i = 2; i < lhs->shape().dimensions_size(); ++i) {
+      reduction_dims.push_back(i);
+      num_elements *= lhs->shape().dimensions(i);
+    }
+    auto flow = builder_.Reduce(lhs, zero,
+                                MakeAddComputation(lhs->shape().element_type()),
+                                reduction_dims);
+    auto denominator = MakeScalar(num_elements, lhs->shape().element_type());
+    std::tie(flow, denominator) = EqualizeShape(flow, denominator);
+    flow = builder_.Divide(flow, denominator);
+    HloTensorType output_shape(flow->shape());
+    while (output_shape.Rank() < lhs->shape().dimensions_size()) {
+      output_shape.AddDimension(1);
+    }
+    return {builder_.Reshape(flow, output_shape)};
+  }
+
+  std::vector<HloFlow> OpSqueeze(const pblczero::NodeProto& node) {
+    CheckKnownAttributes(node, 2, {});
```

Comment:
Like `Split`, before opset 13 it used an "axes" attribute instead of input: https://github.com/onnx/onnx/blob/main/docs/Changelog.md#attributes-189

Reply:
Same as above.